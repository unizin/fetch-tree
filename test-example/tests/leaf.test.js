/* eslint-env jest */
/* eslint-disable react/prop-types */
import React from 'react'

import { renderSnapshot } from './test-utils.js'
import fetchTree from 'fetch-tree'

import Todo from '../src/components/todo'
import Loading from '../src/components/loading'

const LeafNode = fetchTree({
    component: Todo,
    busy: Loading,
})

test(`Render Leaf Node`, async () => {
    const report = await renderSnapshot(
        <LeafNode id={1} />
    )
    expect(report.apiRequests).toEqual({
        fetchTodo: 1,
    })
    expect(report.loadingScreens).toBe(1)
})
