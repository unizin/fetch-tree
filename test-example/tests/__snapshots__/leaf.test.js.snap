exports[`test Render Leaf Node 1`] = `
<div>
  id: 1
  <input
    checked={false}
    type="checkbox" />
  <p>
    Fouti hi rovumjak wuwab ojki utemun ti voblom mefup ivkiz nadahuw ci.
  </p>
</div>
`;
